core/sql/optimizer/NATable.cpp

```diff
@@ -1472,58 +1478,161 @@ static ItemExpr * getRangePartitionBoundaryValuesFromEncodedKeys(
               // value. Note that the buffer has a prefix of some bytes with actual key
               // values, followed by bytes that are zeroed out. 
 
+              // the number of bytes actually provided in the key (not filled in)
+              int numBytesInProvidedVal = encodedKeyLen-valOffset;
+
+              if (nullHdrSize && numBytesInProvidedVal <= 0)
```

Comment:
Couldn't this happen, for example?

currOffset = 10
encodedKeyLen = 11
nullHdrSize = 2
valOffset = 12

In other words, the partially provided key covers only the NULL indicator or part of it.